http-scala-fx/src/main/scala/fx/RetryHttpException.scala

```diff
@@ -0,0 +1,23 @@
+package fx
+
+import java.net.http.HttpResponse
+
+opaque type RetryPolicyHttpException[A] =
+  RetryPolicyHttpException.UnhandledRetryPolicyException |
+    RetryPolicyHttpException.RetriesExhaustedException[A]
+
+object RetryPolicyHttpException:
+  private[fx] final case class UnhandledRetryPolicyException(
+      message: String,
+      cause: Throwable | Null)
+      extends RuntimeException(message, cause)
+  private[fx] final case class RetriesExhaustedException[A](r: HttpResponse[A])(
+      using Show[HttpResponse[A]])
+      extends RuntimeException:
+    override def getMessage: String =
+      s"${summon[Show[HttpResponse[A]]].show(r)} retries exhausted."
```

Comment:
If we provide a name to the implicit parameter, we could simplify the function body a bit:
```suggestion
  private[fx] final case class RetriesExhaustedException[A](r: HttpResponse[A])(
      using S: Show[HttpResponse[A]])
      extends RuntimeException:
    override def getMessage: String = s"${S.show(r)} retries exhausted."
```